salt/orch/update.sls

```diff
@@ -83,6 +83,33 @@ pre-orchestration-migration:
     - require:
       - update-modules
 
+# NOTE: Remove me for 4.0
+#
+# During an upgrade from 2.0 to 3.0, as we go master by master first, the first master will not
+# succeed on the orchestration if it doesn't have an etcd member. Assume M{1,2,3}, W{1,2}. Assume
+# etcd members are running on M2, W1 and W2.
+#
+# M1 updates its configurations on highstate and refers to the etcd nodes with the new names
+# (hostnames) instead of machine-ids, but M2, W1 and W2 still didn't run anything to refresh their
+# certificates and their etcd instances, what will make M1 fail because it cannot connect to any
+# etcd instance (as all certificates look invalid at this point for M2.hostname, W1.hostname and
+# W2.hostname). This makes the apiserver on M1 fail restarting itself until the orchestration reaches
+# M2 [there's no hard dependency on states between masters], but the orchestration already failed on
+# M1, so the global result will be failure nevertheless.
+#
+# Let's force etcd to refresh certificates on all machines, restarting the etcd service so we can
+# continue with the upgrade, as certificates will be valid for the old and the new SAN.
+etcd-setup:
+  salt.state:
+    - tgt: 'roles:etcd'
+    - tgt_type: grain
+    - sls:
+      - etcd
+    - batch: 1
```

Comment:
Any reason for updating them one by one? Couldn't we run this step with `batch: {{ salt.caasp_etcd.get_cluster_size() }}`?

Reply:
It's better to do it safely so we don't restart all etcd instances at once. In any case, the `batch: 1` while being safer has negligible impact in terms of timing overall.

Reply:
That would mean all etcd's could restart at the same time - we want to keep the cluster up and functional during the upgrade.

Reply:
Right.